powerauth-java-server/src/main/java/com/wultra/security/powerauth/app/server/converter/PrivateKeyRegistryDeserializer.java

```diff
@@ -0,0 +1,112 @@
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Iterator;
+
+/**
+ * JSON deserializer for private keys.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Slf4j
+public class PrivateKeyRegistryDeserializer extends JsonDeserializer<PrivateKeyRegistry> {
+
+    private static final KeyConvertor KEY_CONVERTOR_EC = new KeyConvertor();
+    private static final PqcDsaKeyConvertor KEY_CONVERTOR_PQC_DSA = new PqcDsaKeyConvertor();
+
+    @Override
+    public PrivateKeyRegistry deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+        final JsonNode root = jsonParser.getCodec().readTree(jsonParser);
+        final PrivateKeyRegistry keyRegistry = new PrivateKeyRegistry();
+
+        final JsonNode privateKeysNode = root.get("privateKeys");
+        final Iterator<String> algorithmNames = privateKeysNode.fieldNames();
+        while (algorithmNames.hasNext()) {
+            final String algorithmName = algorithmNames.next();
+            final SharedSecretAlgorithm algorithm = SharedSecretAlgorithm.valueOf(algorithmName);
+            final JsonNode algorithmNode = privateKeysNode.get(algorithmName);
+            if (algorithmNode == null || algorithmNode.isEmpty()) {
+                continue;
+            }
+            final Iterator<String> keyTypeNames = algorithmNode.fieldNames();
+            while (keyTypeNames.hasNext()) {
+                final String keyTypeName = keyTypeNames.next();
+                final KeyType keyType = KeyType.valueOf(keyTypeName);
+                final byte[] encodedKey = algorithmNode.get(keyTypeName).binaryValue();
+                final PrivateKey key = deserializePrivateKey(algorithm, keyType, encodedKey);
+                keyRegistry.storePrivateKey(algorithm, keyType, key);
+            }
+        }
+        return keyRegistry;
+    }
+
+    private PrivateKey deserializePrivateKey(SharedSecretAlgorithm algorithm, KeyType keyType, byte[] encodedKey) throws IOException {
+        try {
+            return switch (algorithm) {
+                case EC_P256, EC_P384 -> {
+                    if (keyType != KeyType.ECDSA) {
+                        throw new IOException("Unsupported key type: " + keyType + " for algorithm " + algorithm);
+                    }
+                    yield KEY_CONVERTOR_EC.convertBytesToPrivateKey(getCurve(algorithm), encodedKey);
+                }
+                case EC_P384_ML_L3 -> {
+                    switch (keyType) {
+                        case ECDSA -> {
+                            yield KEY_CONVERTOR_EC.convertBytesToPrivateKey(EcCurve.P384, encodedKey);
+                        }
+                        case MLDSA -> {
+                            yield KEY_CONVERTOR_PQC_DSA.convertBytesToPrivateKey(encodedKey);
+                        }
```

Comment:
Those one liners could be simplified.
```suggestion
                        case MLDSA -> KEY_CONVERTOR_PQC_DSA.convertBytesToPrivateKey(encodedKey);
```